camayoc/qcs_models.py

```diff
@@ -0,0 +1,175 @@
+        return packet
+
+    def to_str(self):
+        """Return the string representation of the model."""
+        return pformat(self.__dict__)
+
+    def __repr__(self):
+        """For `print` and `pprint`."""
+        return self.to_str()
+
+    def __eq__(self, other):
+        """Return true if both objects are equal."""
+        if isinstance(other, HostCredential):
+            return self.__dict__ == other.__dict__
+
+        diffs = 0
+        password_matcher = re.compile(MASKED_PASSWORD_OUTPUT)
+        for key, value in self.fields().items():
+            if key == 'password':
+                if not password_matcher.match(other.get(key)):
+                    diffs += 1
+            else:
+                if not other.get(key) == value:
+                    diffs += 1
+        return diffs == 0
+
+    def __ne__(self, other):
+        """Return true if both objects are not equal."""
+        return not self == other
+
+
+class NetworkProfile(object):
+    """A class to aid in CRUD test cases for network profiles."""
+
+    def __init__(
+            self,
+            name=None,
+            hosts=None,
+            ssh_port=None,
+            credential_ids=None,
+            _id=None):
+        """Objects of this type contain data associated with network profiles.
+
+        Network profiles can be created on the quipucords server by
+        instantiating a NetworkProfile object. A unique name and username are
+        provided by default. In order to create a valid network profile,
+        you must specify at least one existing host credential and one host.
+
+        Example::
+            >>> from camayoc import api
+            >>> from camayoc.qcs_models import NetworkProfile
+            >>> client = api.QCS_Client()
+
+            >>> hostcred = HostCredential()
+            >>> hostcred.password = 'foo'
+            >>> create_response = client.create_host_cred(hostcred.payload())
+            >>> hostcred._id = create_response.json()['id']
+
+            >>> netprof = NetworkProfile(hosts=['0.0.0.0'],
+                                         credential_ids=['host._id'])
+            >>> client.create_net_prof(netprof.payload())
+        """
+        self.name = str(uuid.uuid4()) if not name else name
+        self.hosts = hosts
+        self.ssh_port = 22 if not name else name
+        self.credentials = credential_ids
+        self._id = _id
+
+    def fields(self):
+        """Return a dictionary with all fields."""
+        fields = self.payload()
+        fields['id'] = self._id
+        return fields
+
+    def payload(self):
+        """Return a dictionary for POST or PUT requests.
+
+        Returns proper data for creating host credential with POST to
+        /api/vi/profiles/networks/ or updating host credential with PUT to
+        /api/v1/profiles/networks/{_id}
+
+        """
+        packet = {k: v for k, v in self.__dict__.items() if k != '_id'}
```

Comment:
This may be a bad word to use because of the association with TCP/IP packets. "data" may be better. Other options not to use would be json, because it is a module and payload is the name of the function!